### PR TITLE
Fix - Abbreviated Number - API and validation change

### PR DESCRIPTION
**Pull request by ganeshkumar1989**
MFRWW-227

### Description of the Change

* The abbreviated number for line was validated wrongly to start with 7. This has been fixed to validate with 2.
* The telephony APIs were used instead of the line APIs. This has been fixed.

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build#6120.0 ✘  
* init
  * deps ✘

**Comment by ovh-cds** [comment]
CDS Report build#6120.1 ✘  
* init
  * deps ✘

File: client/app/telecom/telephony/line/calls/abbreviatedNumbers/telecom-telephony-line-calls-abbreviatedNumbers.controller.js [hunk 1]
```diff
@@ -9,21 +9,22 @@ angular.module('managerApp').controller('TelecomTelephonyLineCallsAbbreviatedNum
...
 
   insert(abbreviatedNumber) {
-    return this.OvhApiTelephony.AbbreviatedNumber().v6().insert({
+    return this.OvhApiTelephony.Line().AbbreviatedNumber().v6().insert({
       billingAccount: this.$stateParams.billingAccount,
       serviceName: this.$stateParams.serviceName,
     }, abbreviatedNumber).$promise;
   }
 
   update(abbreviatedNumber) {
```
Comment (JeremyDec) [inline client/app/telecom/telephony/line/calls/abbreviatedNumbers/telecom-telephony-line-calls-abbreviatedNumbers.controller.js]: You could use destructive pattern like you did above with remove method

File: client/components/telecom/telephony/abbreviatedNumbers/telecom-telephony-abbreviated-numbers.modal.html [hunk 2]
```diff
@@ -19,7 +19,7 @@ <h2 class="oui-modal__title" data-ng-bind=":: AbbreviatedNumberModal.title"></h2
         <oui-field
             data-label="{{:: 'telephony_abbreviated_numbers_id' | translate }}"
             data-error-messages="{pattern: AbbreviatedNumberModal.pattern.errorMessage}"
-            data-help-text="{{:: 'telephony_abbreviated_numbers_id_help' | translate }}">
+            data-help-text="{{:: 'telephony_abbreviated_numbers_id_help' | translate: {digit:AbbreviatedNumberModal.pattern.startDigit} }}">
```
Comment (JeremyDec) [inline client/components/telecom/telephony/abbreviatedNumbers/telecom-telephony-abbreviated-numbers.modal.html]: Space after and before brackets : ` {digit:AbbreviatedNumberModal.pattern.startDigit} `

**Comment by ovh-cds** [comment]
CDS Report build-yarn#16.0 ✘  
* Stage 1
  * deps ✔ 
* Stage 2
  * build ✘  
  * test ✔

File: client/components/telecom/telephony/abbreviatedNumbers/translations/Messages_fr_FR.xml [hunk 3]
```diff
@@ -4,7 +4,7 @@
    <translation id="telephony_abbreviated_numbers_write_error" qtlid="334244">Impossible de mettre à jour le numéro abrégé.</translation>
    <translation id="telephony_abbreviated_numbers_write_success" qtlid="334257">Le numéro abrégé {{num}} a été mis à jour.</translation>
    <translation id="telephony_abbreviated_numbers_id" qtlid="334283">Numéro abrégé</translation>
-   <translation id="telephony_abbreviated_numbers_id_help" qtlid="480279">Doit être un nombre à 3 ou 4 chiffres, débutant par un 7.</translation>
+   <translation id="telephony_abbreviated_numbers_id_help" qtlid="480279">Doit être un nombre à 3 ou 4 chiffres, débutant par un {{digit}}.</translation>
```
Comment (marie-j) [inline client/components/telecom/telephony/abbreviatedNumbers/translations/Messages_fr_FR.xml]: Maybe the `qtlid` should be removed. Can you check if you're supposed to keep it in that case ?

File: client/app/telecom/telephony/line/calls/abbreviatedNumbers/telecom-telephony-line-calls-abbreviatedNumbers.controller.js (hunk 1 above)
Comment (ganeshkumar1989) [reply client/app/telecom/telephony/line/calls/abbreviatedNumbers/telecom-telephony-line-calls-abbreviatedNumbers.controller.js]: Do you mean destructuring assignment pattern for the method argument?

Comment (JeremyDec) [reply client/app/telecom/telephony/line/calls/abbreviatedNumbers/telecom-telephony-line-calls-abbreviatedNumbers.controller.js]: Yes i mean it

File: client/components/telecom/telephony/abbreviatedNumbers/telecom-telephony-abbreviated-numbers.modal.html (hunk 2 above)
Comment (ganeshkumar1989) [reply client/components/telecom/telephony/abbreviatedNumbers/telecom-telephony-abbreviated-numbers.modal.html]: This is fixed

File: .gitignore [hunk 4]
```diff
@@ -1,5 +1,6 @@
 node_modules
 client/index.html
+client/bower_components
```
[the remark this answers is not in the document]
Comment (ganeshkumar1989) [reply .gitignore]: This change is done

File: client/app/telecom/telephony/line/calls/abbreviatedNumbers/telecom-telephony-line-calls-abbreviatedNumbers.controller.js [hunk 5]
```diff
@@ -9,21 +9,22 @@ angular.module('managerApp').controller('TelecomTelephonyLineCallsAbbreviatedNum
   }
 
   remove({ abbreviatedNumber }) {
-    return this.OvhApiTelephony.AbbreviatedNumber().v6().remove({
+    return this.OvhApiTelephony.Line().AbbreviatedNumber().v6().remove({
       billingAccount: this.$stateParams.billingAccount,
+      serviceName: this.$stateParams.serviceName,
```
[the remark this answers is not in the document]
Comment (ganeshkumar1989) [reply client/app/telecom/telephony/line/calls/abbreviatedNumbers/telecom-telephony-line-calls-abbreviatedNumbers.controller.js]: This is done :)

File: client/components/telecom/telephony/abbreviatedNumbers/translations/Messages_fr_FR.xml (hunk 3 above)
Comment (ganeshkumar1989) [reply client/components/telecom/telephony/abbreviatedNumbers/translations/Messages_fr_FR.xml]: This is done :)

**Pull request merged**